src/util/atomic_io.h

```diff
@@ -50,4 +50,7 @@ ssize_t sss_atomic_write_safe_s(int fd, void *buf, size_t len);
  */
 ssize_t sss_atomic_read_safe_s(int fd, void *buf, size_t max_len, size_t *_len);
 
+ssize_t sss_atomic_io(int fd, void *buf, size_t n, bool do_read);
```

Comment:
Taking into account that:
 - you don't use "do_read = false" variant,
 - function name is a bit misleading,
 - implementation is pretty much == read()/write() (barring retry on EINTR but that's not important for the use case)
I suggest to not introduce this helper but to use `read()` directly.

Reply:
I mean, if `read()` reads less than requested then next immediate read() will almost certainly return `EWOULDBLOCK` anyway.

Reply:
Replaced in the latest version.